src/org/exist/xquery/functions/fn/LoadXQueryModule.java

```diff
@@ -0,0 +1,214 @@
+                            "function with the given name; its key is the arity of the function, as an xs:integer value, and its associated " +
+                            "value is the function itself, as a function item. The function can be invoked using the rules for dynamic " +
+                            "function invocation.")
+    );
+
+    public final static StringValue OPTIONS_LOCATION_HINTS = new StringValue("location-hints");
+    public final static StringValue OPTIONS_XQUERY_VERSION = new StringValue("xquery-version");
+
+    public final static StringValue RESULT_FUNCTIONS = new StringValue("functions");
+    public final static StringValue RESULT_VARIABLES = new StringValue("variables");
+
+    public LoadXQueryModule(XQueryContext context, FunctionSignature signature) {
+        super(context, signature);
+    }
+
+    @Override
+    public Sequence eval(Sequence[] args, Sequence contextSequence) throws XPathException {
+        final String targetNamespace = args[0].getStringValue();
+        Sequence locationHints = Sequence.EMPTY_SEQUENCE;
+        String xqVersion = getXQueryVersion(context.getXQueryVersion());
+
+        // evaluate options
+        if (getArgumentCount() == 2) {
+            final AbstractMapType map = (AbstractMapType) args[1].itemAt(0);
+            locationHints = map.get(OPTIONS_LOCATION_HINTS);
+
+            final Sequence versions = map.get(OPTIONS_XQUERY_VERSION);
+            if (!versions.isEmpty()) {
+                xqVersion = versions.itemAt(0).getStringValue();
+            }
+        }
+
+        // create temporary context so main context is not polluted
+        final XQueryContext tempContext = new XQueryContext(context.getBroker().getBrokerPool());
+        tempContext.setModuleLoadPath(context.getModuleLoadPath());
+
+        Module module = null;
+        if (locationHints.isEmpty()) {
+            // no location hint given, resolve from statically known modules
+            module = tempContext.importModule(targetNamespace, null, null);
+        } else {
+            // try to resolve the module from one of the location hints
+            for (final SequenceIterator i = locationHints.iterate(); i.hasNext(); ) {
+                final String location = i.nextItem().getStringValue();
+                final Module importedModule = tempContext.importModule(null, null, location);
+                if (importedModule != null && importedModule.getNamespaceURI().equals(targetNamespace) &&
+                        xqVersion.equals(getXQueryVersion(tempContext.getXQueryVersion()))) {
+                    module = importedModule;
+                    break;
+                }
+            }
+        }
+
+        // not found, raise error
+        if (module == null) {
+            throw new XPathException(this, ErrorCodes.FOQM0002, "Module with URI " + targetNamespace + " not found");
```

Comment:
I don't think this is the right error if `xqVersion` doesn't match. However I am not certain from the spec what the error should be.